app/scripts/modules/core/src/serverGroup/details/ServerGroupDetails.tsx

```diff
@@ -1,85 +1,122 @@
-  accountId: string;
-  region: string;
-  serverGroup: string;
-}
-
-export interface IServerGroupDetailsProps {
-  app: Application;
-  serverGroup: {
-    name: string;
-    accountId: string;
-    region: string;
-  };
-}
-
-export interface IServerGroupDetailsState {
-  angular: {
-    template: string,
-    controller: string,
-  };
-  ServerGroupDetailsComponent: React.ComponentClass<IServerGroupDetailsProps>,
-  accountId: string;
-  provider: string;
-  region: string;
-}
+import { IServerGroupDetailsProps, IServerGroupDetailsState } from './ServerGroupDetailsWrapper';
+import { RunningTasks } from './RunningTasks';
+import { ServerGroupInsightActions } from './ServerGroupInsightActions';
 
 @BindAll()
 export class ServerGroupDetails extends React.Component<IServerGroupDetailsProps, IServerGroupDetailsState> {
+  private destroy$ = new Subject();
+
   constructor(props: IServerGroupDetailsProps) {
     super(props);
 
-    const $stateParams: IServerGroupDetailsStateParams = ReactInjector.$stateParams as any;
-
     this.state = {
-      accountId: $stateParams.accountId,
-      provider: $stateParams.provider,
-      region: $stateParams.region,
-      angular: {
-        template: undefined,
-        controller: undefined,
-      },
-      ServerGroupDetailsComponent: undefined,
+      loading: true,
+      serverGroup: undefined,
+      disabledDate: undefined,
     };
   }
 
+  private autoClose(): void {
+    ReactInjector.$state.params.allowModalToStayOpen = true;
+    ReactInjector.$state.go('^', null, { location: 'replace' });
+  }
+
   public componentDidMount(): void {
-    const { provider, accountId } = this.state;
-    const { versionedCloudProviderService } = ReactInjector;
-    $q.all([
-      versionedCloudProviderService.getValue(provider, accountId, 'serverGroup.detailsComponent'),
-      versionedCloudProviderService.getValue(provider, accountId, 'serverGroup.detailsTemplateUrl'),
-      versionedCloudProviderService.getValue(provider, accountId, 'serverGroup.detailsController'),
-    ]).then((values: [React.ComponentClass<IServerGroupDetailsProps>, string, string]) => {
-      const [ component, templateUrl, controller ] = values;
-      const template = templateUrl ? $templateCache.get<string>(templateUrl) : undefined;
-      this.setState({ angular: { template, controller }, ServerGroupDetailsComponent: component });
-    });
+    this.props.detailsGetter(this.props, (state) => this.setState(state), this.autoClose);
+    this.props.app.serverGroups.onRefresh(null, () => this.props.detailsGetter(this.props, (state) => this.setState(state), this.autoClose));
+  }
+
+  public componentWillReceiveProps(nextProps: IServerGroupDetailsProps): void {
+    if (nextProps.serverGroup !== this.props.serverGroup) {
+      this.props.detailsGetter(nextProps, (state) => this.setState(state), this.autoClose);
+    }
+  }
+
+  public componentWillUnmount(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
```

Comment:
I asked in #rx-support and learned that calling `complete()` isn't really necessary for this case